### PR TITLE
feat: add unit tests

### LINKED ISSUE #164: feat: add tests
### Describe the feature you'd like to request

We should add some tests.

### Describe the solution you'd like to see

Setup test tooling for react / unit tests using vitest, and write some/few tests (coverage target will be defined later).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by coxmars**
## Summary 📌

- Closes #164 
- This PR focuses on setting up unit testing and initial test cases for the **ArkMarket** project.

### 1. **Testing Environment Setup**
- Configured **Vitest** as the testing framework, adding it to the project dependencies.
- Created the initial configuration in **`vite.config.ts`** to support testing with `jsdom` environment and `@testing-library/react` for rendering components.
- Added scripts in the **`package.json`** to run the tests, including:
  - `test`: Runs all tests.
  - `test:watch`: Runs tests in watch mode.
  - `test:coverage`: Runs tests and generates code coverage reports.

### 2. **Unit Tests**
- Created a directory **`__tests__`** inside the **`src`** folder to house test files.
- Added unit tests for key components:
  - **`copy-button.test.tsx`**: Tests the functionality of the Copy Button component, including rendering and simulating a button click.
  - **`authentication.test.tsx`**: Verifies the rendering of the Authentication component, including wallet connection, mobile view image, and ArkMarket branding.
  - **`auth-switcher.test.tsx`**: Ensures the `AuthSwitcher` component toggles between login states.
  - **`buy-now-dialog.test.tsx`**: Verifies the functionality of the `Buy Now Dialog` component.
  - **`external-link.test.tsx`**: Tests that external links open correctly.

### 3. **Test Coverage 🥷🏼**
- Added basic unit test cases for the components mentioned above to ensure rendering and user interaction logic work as expected.
- Integrated coverage reporting to track and improve test coverage for components across the project:
  - Configured coverage reports to be generated in **text**, **html**, and **json** formats for better insights.
  
### 4. **Test Results ✅**
- The initial test suite has been executed successfully, with 5 unit tests added to cover some components.

image

### 5. **Pending 🔑**
- I will add more tests for the pending components, I do not know if for all of them but at least 10 more or let me know your preference 🫡

### CODE REVIEW

File: packages/ui/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@ark-market/tsconfig/internal-package.json",
+  // "extends": "@ark-market/tsconfig/internal-package.json",
```
Comment (gershon) [inline packages/ui/tsconfig.json:2]: Is this commented for a specific reason, if not we should keep this tsconfig as it was.

**Comment by gershon** [comment]
@coxmars looks good! there is just an issue with updated packages/ui tsconfig, I'll merge when this is fixed.

**Comment by coxmars** [comment]
> @coxmars looks good! there is just an issue with updated packages/ui tsconfig, I'll merge when this is fixed.

Thanks for the feedback! I noticed the issue with the packages/ui tsconfig. Also, I'd like to add a few more tests for additional components before the merge (today).

Let me know if that works for you! 🫡

**Comment by gershon** [comment]
@coxmars existing react tests are sufficient for now as the goal for now is to have test setup and later decide what to tests. You can try to add unit tests for lib functions like [ownerOrShortAddress.ts](https://github.com/ArkProjectNFTs/ark-market/blob/main/apps/arkmarket/src/lib/ownerOrShortAddress.ts).

Also, removing`packages/ui` tsconfig changes fixes CI lint and typecheck issues.

Lmk if you have issues or need other informations 🚀 .

**Comment by coxmars** [comment]
Hi @gershon, unit test for ownerorShortAddress.ts applied, restored packages/ui tsconfig file and merge applied, I think now it looks good, if you need something else just tell me. Thanks 🫡

**Pull request merged**